### DISCUSSION TITLE
Resize longest side?

### DISCUSSION THREAD

**Discussion by dlambert-personal** [saucecontrol/PhotoSauce#141] [Q&A]
I can see where I can size by Width or Height, but is there a way to size so either Width or Height are sized so the longest side of the image is [x]?

**Comment by saucecontrol** [comment]
Have a look at the [`ResizeMode`](https://docs.photosauce.net/api/PhotoSauce.MagicScaler.ProcessImageSettings.html#PhotoSauce_MagicScaler_ProcessImageSettings_ResizeMode) setting.  With `CropScaleMode.Contain` or `CropScaleMode.Max`, you can set `Width` and `Height` to the same value (your largest desired dimension), and the smaller dimension will be scaled proportionally.  The difference between those two is that `Max` will never enlarge if your input is already smaller than the largest dimension given.

**Reply by dlambert-personal** [reply]
Thanks - that may get me where I need to be.  I may have need to enlarge -- still validating that.

**Reply by dlambert-personal** [reply]
I reviewed my requirements, and unfortunately, I will need to support enlarging sometimes, so it looks like I still need to be able to either find height & width so I can sort which is longest -- all outside MagicImageProcessor.

**Reply by saucecontrol** [reply]
Did you try `CropScaleMode.Contain`?  That sounds like exactly what you're asking for.

If that doesn't cover your needs, you'll need to give some more specific examples of what you're looking for.
